pkg/controller/gitlab/buckets.go

```diff
@@ -154,7 +156,42 @@ func (r *bucketReconciler) reconcile(ctx context.Context) error {
 }
 
 func (r *bucketReconciler) getClaimKind() string {
-	return bucketClaimKind + "-" + r.bucketName
+	return bucketClaimKind + "-" + r.componentName
+}
+
+func (r *bucketReconciler) getHelmValues(ctx context.Context, values map[string]string) error {
+	return r.baseResourceReconciler.loadHelmValues(ctx, values, r.helmValuesFunction)
+}
+
+const (
+	helmValueBucketFmt           = "global.appConfig.%s.bucket"
+	helmValueConnectionSecretFmt = "global.appConfig.%s.connection.secret"
+	helmValueConnectionKeyFmt    = "global.appConfig.%s.connection.key"
+	helmValueTaskRunnerSecret    = "gitlab.task-runner.backups.objectStorage.config.secret"
+	helmValueTaskRunnerKey       = "gitlab.task-runner.backups.objectStorage.config.key"
+	helmValueBackupsTempBucket   = "global.appConfig.backups.tmpBucket"
+)
+
+// bucketConnectionHelmValues map of helm set key/value paris
+// https://docs.gitlab.com/charts/advanced/external-object-storage/index.html#lfs-artifacts-uploads-packages-external-diffs-pseudonymizer
+func bucketConnectionHelmValues(values map[string]string, name string, secret *corev1.Secret) {
+	values[fmt.Sprintf(helmValueBucketFmt, name)] = string(secret.Data[xpcorev1alpha1.ResourceCredentialsSecretEndpointKey])
+	values[fmt.Sprintf(helmValueConnectionSecretFmt, name)] = secret.GetName()
```

Comment:
do we know that when these funcs are called the secret pointer will be non nil? same on line 187 and in other implementors of this `helmValuesFunction` interface

Reply:
In theory, yes, the secret can be nil, thus it will throw (and justifiably so) `nil reference` error. 
In practice, however, this is called from: